### PR TITLE
Upgrade to TF 0.12

### PR DESCRIPTION
**Pull request by stefansedich**
Upgrading to TF 0.12, depends on https://github.com/cloudposse/terraform-aws-route53-cluster-hostname/pull/13 being merged too as this module references that one.

### CODE REVIEW

**Comment by stefansedich** [comment]
I explicity set the `TERRAFORM_VERSION` in the Makefile so that the build would use the right version, not sure if this is the desired way to do it but seems to be the easiest way without making changes to the base build stuff.

However it looks like build harness calls `terraform validate` with `-check-variables=false` which is no longer a valid argument to validate.

**Comment by stefansedich** [comment]
Ok got a bit further with the build and now have it in a state where it would pass if we merged https://github.com/cloudposse/terraform-aws-route53-cluster-hostname/pull/14 and created a release, I could then update the tag ref in this PR and everything should be green.

File: main.tf [hunk 1]
```diff
@@ -1,52 +1,52 @@
 # Define composite variables for resources
 module "label" {
   source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.3.5"
```
Comment (aknysh) [inline main.tf]: update to https://github.com/cloudposse/terraform-null-label/releases/tag/0.12.0, it's for TF 0.12

**Comment by stefansedich** [comment]
Thanks @aknysh terraform-null-label is updated, will away the other PR to get merged then do the final change to this PR to bump the terraform-aws-route53-cluster-hostname ref.

**Comment by stefansedich** [comment]
@aknysh @osterman I am assuming you guys will tackle this with your own PR like you did for the terraform-aws-route53-cluster-hostname module? will this one be something you will get to soon too?

**Comment by aknysh** [comment]
@stefansedich yes, we are working on converting all our modules.
we are also adding tests to ensure that the conversion produces usable modules w/o any issues.

**Comment by stefansedich** [comment]
Ok cool thanks @aknysh I will close this PR, do you have any idea how long it might be until you get to this module?

**Pull request closed without merging**